### PR TITLE
[terra-date-time-picker] Fix to clear out invalid date value on clicking time input

### LINKED ISSUE #1217: [terra-date-time-picker] Focusing on Time Input does not clear Invalid default value
# Bug Report

## Description

When an **Invalid default date and time value** is specified, then focusing on the "Time Input", only clears the time input and not the date value.

## Steps to Reproduce

1. Goto https://engineering.cerner.com/terra-framework/tests/terra-date-time-picker/date-time-picker/date-time-picker-default-date-excluded
2. Click on the `hour/minute` input of the time input.
3. Notice that only the time input value gets cleared and not the date input value.
4. Whereas on refreshing and focusing on the date input, clears both the date and time input values.

Similarly can also be reproduced on https://engineering.cerner.com/terra-framework/tests/terra-date-time-picker/date-time-picker/date-time-picker-default-date-out-of-range

## Expected Behavior

Focusing on the "Time Input" should clear the entire date and time value when an invalid **default** date and time value is specified.

### PR DESCRIPTION
**Pull request by saket2403**
### Summary
Fix to clear out invalid date value on clicking time input. Changes made in date-picker to set date value to null if value is invalid, instead of retaining previous value.

Closes #1217 

### Deployment Link

https://terra-framew-date-time--pwapwr.herokuapp.com/components

### Testing
Updated wdio snapshots for terra-date-picker and date-time-picker.

### Additional Details

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-date-picker/tests/wdio/date-picker-spec.js [hunk 1]
```diff
@@ -457,19 +457,6 @@ Terra.describeViewports('Date Picker', ['medium'], () => {
     });
 
     Terra.it.matchesScreenshot('ordinal date not set');
-
-    it('Open the date picker', () => {
```
Comment (emilyrohrbough) [inline packages/terra-date-picker/tests/wdio/date-picker-spec.js:461]: Why are these tests removed? Did we need to add an additional scenario to validate this fix?

Comment (saket2403) [reply packages/terra-date-picker/tests/wdio/date-picker-spec.js:461]: When we open the calendar after clearing out the date, today's date will be in bold which will change evereyday so I removed the test.

Comment (emilyrohrbough) [reply packages/terra-date-picker/tests/wdio/date-picker-spec.js:461]: @saket2403 Can we update the test to validate this interaction without relying on a screenshot?

Comment (saket2403) [reply packages/terra-date-picker/tests/wdio/date-picker-spec.js:461]: I can use mockdate to fix the current date and then take a screenshot, something similar to what will be done in #1202 ?

Comment (emilyrohrbough) [reply packages/terra-date-picker/tests/wdio/date-picker-spec.js:461]: Do we need to take a screenshot to validate this is working correctly ?

Comment (saket2403) [reply packages/terra-date-picker/tests/wdio/date-picker-spec.js:461]: We do take a screenshot after clearing the value, should be enough I guess.

**Comment by supreethmr** [comment]
Date is clearing out only when date is an excluded date OR when date is out of range OR when invalid date passed as initial date value. but not clearing the date when invalid date is being typed.

date_picker_clears

**Comment by saket2403** [comment]
> Date is clearing out only when date is an excluded date OR when date is out of range OR when invalid date passed as initial date value. but not clearing the date when invalid date is being typed.
> 
After Splitting the date picker it wont allow wrong values to be entered for date so this should not be an issue

**Comment by benbcai** [comment]
> Date is clearing out only when date is an excluded date OR when date is out of range OR when invalid date passed as initial date value. but not clearing the date when invalid date is being typed.
> 

@supreethmr The term `invalid` is sometimes used loosely. For the sake of this conversation, I'll use the following terms.

- `complete` - A full date value that conforms to the date format.
- `invalid` - A full complete date but it is either excluded or out of range.
- `malformed` - A date value that is not complete or does not conform to the format. (e.g. `293-32-34`)

The clearing logic applies only for complete default date that is invalid (excluded or out-of-range). It does not get cleared after the initial render. However, it is cleared as soon as you attempt to interact with the component (i.e. receives focus or opening the picker). If the default value is malformed, it is ignored and does not get display in the input.

**Review by pranav300 — APPROVED** [review]

**Review by lokesh-0813 — APPROVED** [review]

**Review by avinashg1994 — APPROVED** [review]

**Comment by PaulSolDev** [comment]
Functional review complete. No issues identified.

**Pull request merged**